### DISCUSSION TITLE
Ionic run cmd from ext fails with Error, Run cmd from Android studio Runs

### DISCUSSION THREAD

**Discussion by will-summerlin** [ionic-team/capacitor#7010] [General]
I have a vue app using VScode ide that i am using capacitor to make a hybrid app. I set it up tper instructions, but  when trying to execute the run cmd from vscode i receive this
![image](https://github.com/ionic-team/capacitor/assets/127124390/9ffdd2e0-d0ba-444a-a301-09e095871c40)

 What went wrong:
        Execution failed for task ':capacitor-android:compileDebugJavaWithJavac'.
        > Could not resolve all files for configuration ':capacitor-android:androidJdkImage'.
        > Failed to transform core-for-system-modules.jar to match attributes {artifactType=_internal_android_jdk_image, org.gradle.libraryelements=jar, org.gradle.usage=java-runtime}.
        > Execution failed for JdkImageTransform: C:\Users\wills\AppData\Local\Android\Sdk\platforms\android-33\core-for-system-modules.jar.
        > Error while executing process C:\Program Files\Java\jdk-21\bin\jlink.exe with arguments {--module-path C:\Users\wills\.gradle\caches\transforms-3\4f7d45b6a0a60abeff674c6e3ec3239c\transformed\output\temp\jmod --add-modules java.base --output C:\Users\wills\.gradle\caches\transforms-3\4f7d45b6a0a60abeff674c6e3ec3239c\transformed\output\jdkImage --disable-plugin system-modules}
        
        * Try:
        > Run with --stacktrace option to get the stack trace.
        > Run with --info or --debug option to get more log output.
        > Run with --scan to get full insights.
        
        * Get more help at https://help.gradle.org
        
        BUILD FAILED in 9s62 actionable tasks: 4 executed, 58 up-to-date
        
        
[error] npx cap run android --target=emulator-5554  --external --no-sync --public-host=172.25.0.1 Failed
[error] Android Failed.

but when executing the run command from the linked android studio build it runs. Any clue as to why? 

Summary : npx cap run android fails with above error, run app cmd in android studio runs successfully

**Comment by ivanbacher** [comment]
Any luck? I am having the same issue. https://github.com/ionic-team/capacitor/discussions/7053

**Comment by jcesarmobile** [comment]
for using jdk 21 you need to use gradle 8.4, which is not officially supported for android development yet
for android development jdk 17 is recommended

**Reply by chitgoks** [reply]
perhaps 8.5 will do? ive been deploying android apps with gradle 8.5 and jdk 21. one thing confused me though. if i deploy it using android studio, it works ok (gradle 8.5, agp 8.2.0, jdk 21). but the cli shows that even, even if i used the default gradle settings as well as jdk 17.

**Reply by jcesarmobile** [reply]
For using gradle 8.4 you should use gradle plugin 8.3, which is in beta stage.
For using gradle 8.6 you should use gradle plugin 8.4, which is in alpha stage.

I can't find information about gradle 8.5, but I would imagine it would also require gradle plugin 8.4.

With other versions it might or might not work, but it's not recommended to use different versions than the recommended. 
https://developer.android.com/build/releases/gradle-plugin#updating-gradle

**Comment by will-summerlin** [comment]
The issue was using JDK 21 which is unsupported by capacitor, you must use JDK 17 or lower with Capacitor5
